src/main/java/wooteco/subway/section/domain/Sections.java

```diff
@@ -0,0 +1,193 @@
+package wooteco.subway.section.domain;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import wooteco.subway.exception.DeleteSectionException;
+import wooteco.subway.exception.DuplicatedSectionException;
+import wooteco.subway.exception.InvalidSectionsException;
+import wooteco.subway.exception.NotAddSectionException;
+import wooteco.subway.exception.NotContainStationsException;
+import wooteco.subway.exception.NotFoundTerminalStationException;
+import wooteco.subway.exception.NotRemoveSectionException;
+import wooteco.subway.station.domain.Station;
+
+public class Sections {
+
+    private final List<Section> sections;
+
+    public Sections(List<Section> sections) {
+        if (sections.isEmpty()) {
+            throw new IllegalArgumentException();
+        }
+        this.sections = sortedSections(sections);
+    }
+
+    private List<Section> sortedSections(List<Section> sections) {
+        List<Section> sortedSection = new ArrayList<>();
+        Station currentStation = upTerminalSection(sections).getUpStation();
+
+        for (int i = 0; i < sections.size(); i++) {
+            currentStation = addSection(sections, sortedSection, currentStation);
+        }
+        return sortedSection;
+    }
+
+    private Station addSection(List<Section> sections, List<Section> sortedSections,
+        Station currentStation) {
+        for (Section section : sections) {
+            if (section.isEqualsUpStation(currentStation)) {
+                sortedSections.add(section);
+                return section.getDownStation();
+            }
+        }
+        throw new InvalidSectionsException();
+    }
+
+    private Section upTerminalSection(List<Section> sections) {
+        List<Long> upStationIds = stationIds(sections, true);
```

Comment:
메소드에서 boolean 값으로 구분하기 보단 stream을 이용하거나 각각 메소드 분리 하는건 어떠신가요? 

```suggestion
        List<Long> upStationIds = sections.stream()
                .map(Section::getUpStationId)
                .collect(Collectors.toList());
```

Reply:
upTerminalSection 메서드 길이가 길어져서 각각 메소드로 분리해봤습니다!!
메서드의 인자로 주어지는 boolean 값으로 로직 작동을 분기시키는 방법은 가독성 측면에서 안좋다고 생각했습니다!